api/models/orderDao.js

```diff
@@ -0,0 +1,144 @@
+  return orderNumber;
+};
+
+const order = async (userId, cartId, totalPrice, paymentMethod) => {
+  const orderNumber = createOrderNumber();
+  const queryRunner = await myDataSource.createQueryRunner();
+  await queryRunner.connect();
+  await queryRunner.startTransaction();
+  try {
+    const pay = await queryRunner.query(
+      `
+      UPDATE
+        users
+      SET
+        point=point-?
+      WHERE
+        id=? `,
+      [totalPrice, userId]
+    );
+    const orderTable = await queryRunner.query(
+      `
+      INSERT INTO
+        orders(user_id,status_id,order_number)
+      VALUES (?,1,?);`,
+      [userId, orderNumber]
+    );
+    const orderTableId = orderTable.insertId;
+    const paymentTable = await queryRunner.query(
+      `
+      INSERT INTO
+      payments(order_id,total_price,method_id)
+      VALUES(?,?,?);
+      `,
+      [orderTableId, totalPrice, paymentMethod]
+    );
+    const itemsTable = await queryRunner.query(
+      `INSERT INTO order_items(order_id,item_id,quantity)
+      SELECT ?,item_id,quantity FROM carts WHERE id IN (?);`,
+      [orderTableId, cartId]
+    );
+    const [itemsTableStartId] = await queryRunner.query(
+      `
+      SELECT last_insert_id() as id;
+      `
+    );
+    let itemsIdArray = [];
+    const cartIdLength = typeof cartId == "string" ? 1 : cartId.length;
+    for (let i = 0; i < cartIdLength; i++) {
+      itemsIdArray.push(Number(itemsTableStartId.id) + i);
+    }
+    if (cartIdLength > 1) {
+      for (let i = 0; i < itemsIdArray.length; i++) {
+        const itemsOptionTable = await queryRunner.query(
+          `INSERT INTO order_item_options(order_item_id,option_id)
+      SELECT ?,option_id FROM cart_item_options WHERE cart_item_id=?;  `,
+          [itemsIdArray[i], cartId[i]]
+        );
+      }
+    } else {
+      const itemsOptionTable = await queryRunner.query(
+        `INSERT INTO order_item_options(order_item_id,option_id)
+      SELECT ?,option_id FROM cart_item_options WHERE cart_item_id=?;  `,
+        [itemsIdArray[0], cartId]
+      );
+    }
+    const deleteItemsTable = await queryRunner.query(
+      `
+      DELETE FROM carts
+      WHERE id IN (?);`,
+      [cartId]
+    );
+    const deleteItemsOptionTable = await queryRunner.query(
+      `DELETE FROM cart_item_options WHERE cart_item_id IN (?)`,
+      [cartId]
+    );
+    await queryRunner.commitTransaction();
+    await queryRunner.release();
+    return;
+  } catch (error) {
+    await queryRunner.rollbackTransaction();
+    await queryRunner.release();
+    const err = new Error("GETTING TRANSACTION FAILED ");
+    err.statusCode = 400;
+    throw err;
+  }
+};
+
+const loadOrderStatus = async (userId, oiOrderId) => {
+  const result = await myDataSource.query(
+    `
+      SELECT
+        os.id AS osId,
+        os.status AS osStatus,
+        o.order_number AS oOrderNumber,
+        u.id AS uId,
+        oi.order_id AS oiOrderId
```

Comment:
컬럼명 바꿀 때 다음과 같이 더 명확하게 해주면 좋을 것 같아요.
바뀐 컬럼명을 가지고 결국 코드에서 사용 할 텐데 의미가 명확한 것이 좋습니다.
```suggestion
        os.id AS orderStatusId,
        os.status AS orderStatus,
        o.order_number AS OrderNumber,
        u.id AS userId,
        oi.order_id AS orderId
```